lib/paperclip/io_adapters/data_uri_adapter.rb

```diff
@@ -11,9 +13,32 @@ def initialize(target_uri)
 
     def extract_target(uri)
       data_uri_parts = uri.match(REGEXP) || []
-      StringIO.new(Base64.decode64(data_uri_parts[2] || ''))
+      decoded = get_image_from_match(data_uri_parts)
+
+      if is_svg_image?(decoded)
+        decoded = generate_xml_string(decoded)
+      end
+
+      StringIO.new(decoded)
+    end
+
+    def is_svg_image?(img)
+      img.start_with?('<svg')
     end
 
+    def get_image_from_match(match)
+      return '' if match.length == 0
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.